pkgs/by-name/bu/bun/hooks/default.nix

```
@@ -0,0 +1,86 @@
+{
+  lib,
+  stdenvNoCC,
+  jq,
+  moreutils,
+  cacert,
+  makeSetupHook,
+  bun,
+}:
+
+{
+  fetchDeps =
+    {
+      hash ? "",
+      pname,
+      workspaces ? [ ],
+      preInstall ? "",
+      installFlags ? [ ],
+      ...
+    }@args:
```

```suggestion
  fetchDeps = lib.makeOverridable (
    lib.fetchers.withNormalizedHash { } (
      {
        name ?
          lib.optionalString (args ? "pname" && args ? "version") "${args.pname}-${args.version}-"
          + "bun-deps",
      workspaces ? [ ],
      preInstall ? "",
      installFlags ? [ ],
      outputHash,
      outputHashAlgo,
      ...
    }@args:
```
This introduces a few things:

1) The ability for users to easily override variables in calls of `fetchDeps`, such as `hash`, `src`, `pname`, `version`, etc

2) Proper `name` handling by taking `version` into account, as well as giving us a default fallback in only `bun-deps`

2) Removes the needs for any messing around with different hash types here. `withNormalizedHash` does it all for us, and we only need to take in the direct `mkDerivation` arguments `outputHash` and `outputHashAlgo`